### ISSUE TITLE
Modulator

### ISSUE THREAD

**Issue by gkasprow** [sinara-hw/modulator#1]
For the CERN anti-matter trap, we need to create an amplitude modulated HV RF signal.
Urukul amplitude control seems to be not precise enough to generate; moreover, the resonator Q factor may make such source modulation useless. The plan is to make a simple modulator card that takes the copy of the source signal, multiplies it by Fastino output, and adds it to 500V RF driving the trap. We need a few V of amplitude modulation on top of 500V carrier.
So the plan is to make a dedicated, 2-channel modulator card in EEM form factor. The modulator realizes the function
![obraz](https://user-images.githubusercontent.com/4325054/178995262-b0448f25-6456-4202-979c-598def4981b4.png)
I want to make the design as universal as possible, useful also for other applications
The spec is as follows:
- [ADL5391](https://www.analog.com/media/en/technical-documentation/data-sheets/ADL5391.pdf) multiplier chip with 50R matched single-ended to differential amplifiers
- +/-1V input signal X
- +/-1V / +/-10V modulation signal (Y); 50 or 500Ohm matched
- 2GHz BW
- +/-10V offset signal (Z)
- optional output booster (dedicated output) that boosts the signal to +/- 6V, I will use probably [THS3217](https://www.ti.com/lit/ds/symlink/ths3217.pdf?ts=1657806011053&ref_url=https%253A%252F%252Fwww.google.com%252F)

The question is if such a module would be useful in a quantum lab.

**Comment by jordens** [comment]
Also adding things after a (Paul trap) resonator is tricky to do without impacting Q or resonance. Paul trap people (those with resonators) usually just add modulation to some other electrode as it doesn't need to be large nor calibrated. Might be interesting for Penning traps though. In general this would have a large overlap with Phaser, especially with things like the new 16-tone + shaper architecture which has precisely such a modulator integrated in digital domain.

**Comment by dnadlinger** [comment]
We are using an ADL5391 PCB like this for "poor-man's pulse shaping" of multi-tone RF signals (AOM drive) as controlled by a TTL pulse. In that case, we've put a comparator and switchable active filter onto the PCB to turn the edges of the square pulse into something approximately Gaussian (with switchable bandwidth), but that could have been a simpler add-on board to a more general-purpose card.

Something like this would have been used by many people doing laser-based gates of some sort (incl. for optical power stabilisation), although Phaser with ADC feedback will hopefully provide a nicely integrated solution for all of this at last. For that use case, we aren't interested in DC on the carrier signal, though, but just the usual AOM range, maybe 60 MHz–300 MHz. On that PCB I mentioned, we AC-coupled X and W like suggested by the datasheet and didn't bring out Z (but instead alpha to keep the option for closed-loop gain adjustments).

Having a ready-made ADL5391 card with all the PSU stuff already taken care of (12 V input like the other Sinara stuff) and a nice RF datasheet (S21, noise) would probably be useful.

**Comment by gkasprow** [comment]
@jordens we use 1MHz signal and need DC-coupling so adding of modulated signal may not be that hard. Anyway, we will try both approaches :)

**Comment by dhslichter** [comment]
We have used ADL5391 eval boards in various applications for simple pulse shaping.  Is what is being proposed here basically just a Sinara version of such an eval board, with power supplies etc, and the X, Y, Z exposed as front panel connectors?  Or are you looking at having a Fastino DAC onboard to generate the Y, or other ARTIQ-tied features?

**Comment by gkasprow** [comment]
No, just packaged ADL5391. There is no reason to use Fastino-like DAC since we already use Fastino in the experiment and have free channels. Moreover, I want to make it universal and capable of working with faster DACs.

**Comment by dhslichter** [comment]
This sounds like a simple board that would have various uses around the lab for us.  Do you think you could put two separate channels (two ADL5391) on a single card?  Seems like one would need X, Y, Z inputs and W output per modulator, plus optional GADJ input (to control alpha) which could be on an MMCX (for example).  If you did it that way, it would be 8 SMAs to get two modulators onboard, plus two MMCX that could be connected out another way if one wanted to use them (guessing most users would be content without controlling the gain adjustment).  

What about the need to have the input signals be differential -- do you plan to use fast single-ended-to-differential op amps, or baluns, or have pads to choose which option is used?  Certainly the differential op-amps as an option would be preferred on my end because often one wants to have at least one of the inputs be dc-coupled (for example if using the multiplier for pulse shaping or in a feedback loop).

**Comment by gkasprow** [comment]
the plan is to have both opamps and baluns as an assembly option.
I need two modulators per card.
I was thinking about using 4 SMAs for X and W and MCX for Z, Y and G

**Comment by dnadlinger** [comment]
> I was thinking about using 4 SMAs for X and W and MCX for Z, Y and G

That sounds good to me. From the use cases I've had here so far, both Z and GADJ could probably be dropped to keep the BOM cost low, but I can see how they might be good to have for flexibility.

As for population options, X/W on baluns and Y on a DC-coupled op-amp circuit would probably be a good default for many laser-related ion trap applications.

**Comment by dhslichter** [comment]
I like the idea of having MCX for Z and GADJ, with jumpers/switches on the board to set to "sane" values for those who don't care about them.

> As for population options, X/W on baluns and Y on a DC-coupled op-amp circuit would probably be a good default for many laser-related ion trap applications.

Agree that this is a good default population scheme.

**Comment by gkasprow** [comment]
Would additional ADC/DAC over EEM make sense for setting the offset and gain and measuring the offset?

**Comment by dhslichter** [comment]
Do you mean putting a small ADC/DAC on the board to set/measure Z and GADJ and having that talk over EEM?  Feels like feature creep to me.  I think most use cases in our group we would just use Z=0 and G=1.  I think simplicity is an advantage here, and people can always apply their own voltages to Z and GADJ if they desire, using some coax connector.

**Comment by dhslichter** [comment]
Note if one has W on a balun it will strip any DC offset applied through the Z port anyway.  I think the simple thing is the best thing here -- just default to leaving GADJ open circuit (gain = 1) and Z ports shorted to each other (differential Z = 0 V).

**Comment by gkasprow** [comment]
OK :)
